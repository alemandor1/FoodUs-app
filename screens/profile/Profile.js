import React, { useState, useRef, useEffect } from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { Button } from 'react-native-elements';
import { useNavigation } from '@react-navigation/native'
//import Toast from 'react-native-easy-toast'

import { closeSession, getCurrentUser } from '../../utils/actions';
import Loading from '../../components/Loading';
import InfoUser from '../../components/account/InfoUser';
import AccountOptions from '../../components/account/AccountOptions';

export default function Profile() {
    const toastRef = useRef()
    const navigation = useNavigation()

    const [loading, setLoading] = useState(false)
    const [loadingText, setLoadingText] = useState("")
    const [user, setUser] = useState(null)
    const [reloadUser, setReloadUser] = useState(false)

    useEffect(() => {
        setUser(getCurrentUser())
        setReloadUser(false)
    }, [reloadUser])

    return (
        <View style={styles.container}>
            {
                user && (
                    <View style={{paddingTop: "20%"}}>
                        <InfoUser
                            user={user}
                            setLoading={setLoading}
                            setLoadingText={setLoadingText}
                        />
                        <AccountOptions
                        user={user}
                        setReloadUser={setReloadUser}
                        />
                    </View>
                    )
            }
            <Button
                title="Volver"
                buttonStyle={styles.btnCloseSession}
                titleStyle={styles.btnCloseSessionTitle}
                onPress={() => navigation.navigate("Configuration")}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container : {
        minHeight: "100%",
        backgroundColor: 'white'
    },
    btnCloseSession : {
        marginTop: 30,
        borderRadius: 5,
        backgroundColor: "#F4F5A9",
        borderTopWidth: 1,
        borderTopColor: "#F4F5A9",
        borderBottomWidth: 1,
        borderBottomColor: "#F4F5A9",
        paddingVertical: 10,
        width: "50%",
        alignSelf: "center"
    },
    btnCloseSessionTitle : {
        color: "#000000"
    }
})