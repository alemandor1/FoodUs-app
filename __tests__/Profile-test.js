import React from "react";
import Profile from "../screens/profile/Profile";
import {create} from 'react-test-renderer'
import {render} from "@testing-library/react-native";
import renderer from 'react-test-renderer';

//test snapshot
const tree = create(<Profile />)
test('snapshot', () => {
    expect(tree).toMatchSnapshot();
}) 

//test renderizaci√≥n
describe("<Profile />", () => {
    beforeEach(() => {
        component = render(<Profile />);
    });

    it("Renderiza correctamente", () => {
        expect(component).toBeDefined();
        expect(component.getByTestId("buttonReturn")).toBeDefined();
    });
})

//test unitario
describe('<Profile />', () => {
    it('has 1 child', () => {
      const tree = renderer.create(<Profile />).toJSON();
      expect(tree.children.length).toBe(1);
    });
  });