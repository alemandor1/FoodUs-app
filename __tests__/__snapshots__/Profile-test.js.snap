// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="buttonReturn"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#F4F5A9",
            "borderBottomColor": "#F4F5A9",
            "borderBottomWidth": 1,
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderTopColor": "#F4F5A9",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 30,
            "padding": 8,
            "paddingVertical": 10,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Return
        </Text>
      </View>
    </View>
  </View>
</View>
`;
